src/wombats/lib/wombat_lib.clj

```diff
@@ -0,0 +1,232 @@
+      "w" #(and (= (:y self) (:y %)) (>= shot-range (mod (- (:x self) (:x %)) arena-size)))
+      #(false)))
+    (let [shootable (filter shootable (filter-arena arena "zakano" "wombat"))]
+      (not (empty? (filter #(not (and (= (:x %) (:x self)) (= (:y self) (:y %)))) shootable)))))
+  ([dir arena arena-size shot-range] (can-shoot-enemy? dir arena arena-size shot-range {:x 3 :y 3})))
+
+(defn can-shoot-barrier?
+  "Returns true if there is a barrier within shooting range"
+  ([dir arena arena-size shot-range self]
+    (def shootable (case dir
+      "n" #(and (= (:x self) (:x %)) (>= shot-range (mod (- (:y self) (:y %)) arena-size)))
+      "e" #(and (= (:y self) (:y %)) (>= shot-range (mod (- (:x %) (:x self)) arena-size)))
+      "s" #(and (= (:x self) (:x %)) (>= shot-range (mod (- (:y %) (:y self)) arena-size)))
+      "w" #(and (= (:y self) (:y %)) (>= shot-range (mod (- (:x self) (:x %)) arena-size)))
+      #(false)))
+    (let [shootable (filter shootable (filter-arena arena "wood-barrier" "steel-barrier"))]
+      (not (empty? (filter #(not (and (= (:x %) (:x self)) (= (:y self) (:y %)))) shootable)))))
+  ([dir arena arena-size shot-range] (can-shoot-barrier? dir arena arena-size shot-range {:x 3 :y 3})))
+
+(defn possible-points
+  "Get all locations with possible points"
+  ([arena self]
+    (remove #(and (= (:x %) (:x self)) (= (:y %) (:y self)))
+            (filter-arena (add-locs arena) "food" "wood-barrier" "steel-barrier" "zakano" "wombat")))
+  ([arena]
+    (possible-points arena {:x 3 :y 3})))
+
+(defn build-resp
+  "Helper method to construct the return command"
+  ([action direction]
+    {:action (keyword action)
+     :metadata {:direction (keyword direction)}})
+  ([action] {:action (keyword action)
+             :metadata {}}))
+
+(defn new-direction
+  "Pick new direction to turn to get to loc. If no direction is possible, turns left"
+  [dir loc self arena-half]
+  (def ^:private orientations ["n" "e" "s" "w"])
+  (let [available (remove #(= % dir) orientations)
+        positions (filter #(facing? % loc arena-half self) available)]
+    (if (not (empty? positions))
+        (turn-to-dir dir (first positions))
+        ;TODO improve this logic
+        :left)))
+
+(defn front-tile
+  "Returns a map containing {:x x, :y y}, where x and y are the coordinates directly in front"
+  ([dir arena-size self]
+    (case dir
+      "n" {:x (:x self) :y (mod (dec (:y self)) arena-size)}
+      "e" {:x (mod (inc (:x self)) arena-size) :y (:y self)}
+      "s" {:x (:x self) :y (mod (inc (:y self)) arena-size)}
+      "w" {:x (mod (dec (:x self)) arena-size) :y (:y self)}))
+  ([dir arena-size] front-tile dir arena-size {:x 3 :y 3}))
+
+(defn is-clear?
+  "Return true if you can move forward without a collision or poison"
+  [arena {x :x y :y}]
+  (not (in? (get-in (nth (nth arena y) x) [:contents :type])
+            ["zakano" "wombat" "wood-barrier" "steel-barrier" "poison"])))
```

Comment:
Might be good for these strings to be constants up that can be used in a user's own functions / used here. Maybe something like `zakano-key`, etc.